backend/src/test/java/org/example/backend/service/ProductServiceTest.java

```diff
@@ -32,11 +33,24 @@ void getAllProducts_shouldReturnEmptyList_whenCalledInitially() {
     @Test
     void getProductById_shouldReturnProduct_whenCalledById() {
```

Comment:
in deinem Service-test fehlt der test, ob bei einem Fehler eine exception geworfen wird.
``` @Test
    void ..._shouldThrowException_WhenWentWrong() {
        // GIVEN
        ...Dto newPatient = new ...Dto(...)
        // Mock-Objekt so konfigurieren, dass es eine Ausnahme wirft. z.B 
        when(mockRepository.save(any(...Dto.class))).thenThrow(new NullPointerException("Error message"));
        // WHEN & THEN
        try {
           ...Service.create...(new...);
            verify(mockRepository).save(any(model.class));
            fail("Expected exception, but was not thrown");
        } catch (NullPointerException e) {
            assertEquals("Error message", e.getMessage());
        }
    }```